File: TikraqApp/src/app/model/Leccion.ts
import {Progreso} from './Progreso';
import {Curso} from './Curso';
import {Pregunta} from './Pregunta';

export class Leccion {
  id: number;
  titulo: string;
  contenido: string;
  cursosCursoid: { id:number };

  constructor(id: number=0, titulo: string, contenido: string, cursosCursoid: { id:number }) {
    this.id = id;
    this.titulo = titulo;
    this.contenido = contenido;
    this.cursosCursoid = cursosCursoid;
  }

}

File: Tikraq/src/main/java/com/example/tikraq/entities/Leccion.java
package com.example.tikraq.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "lecciones")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Leccion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "leccionid")
    private Integer id;

    private String titulo;

    private String contenido;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "cursos_cursoid")
    private Curso cursosCursoid;

    @OneToMany(mappedBy = "leccionesLeccionid", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Progreso> progresos = new LinkedHashSet<>();

    @OneToMany(mappedBy = "leccionesLeccionid", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Pregunta> preguntas = new LinkedHashSet<>();
}
